src/modules/container.rs

```diff
@@ -0,0 +1,181 @@
+use super::{Context, Module};
+
+#[cfg(not(target_os = "linux"))]
+pub fn module<'a>(_context: &'a Context) -> Option<Module<'a>> {
+    None
+}
+
+#[cfg(target_os = "linux")]
+pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
+    use super::RootModuleConfig;
+    use crate::configs::container::ContainerConfig;
+    use crate::formatter::StringFormatter;
+    use std::io::BufRead;
+
+    pub fn container_name(context: &Context) -> Option<String> {
+        use crate::utils::context_path;
+
+        if context_path(context, "/proc/vz").exists() && !context_path(context, "/proc/bc").exists()
+        {
+            // OpenVZ
+            return Some("OpenVZ".into());
+        }
+
+        if context_path(context, "/run/host/container-manager").exists() {
+            // OCI
+            return Some("OCI".into());
+        }
+
+        if context_path(context, "/run/systemd/container").exists() {
+            // systemd
+            return Some("Systemd".into());
+        }
+
+        let container_env_path = context_path(context, "/run/.containerenv");
+
+        if container_env_path.exists() {
+            // podman and others
+
+            let image_res = std::fs::File::open(container_env_path)
```

Comment:
Not a blocker but it is recommended to use  [`utils::read_file`](https://github.com/starship/starship/blob/master/src/utils.rs#L13) for getting the contents of a file unless there is a very specific reason not to.

Reply:
fixed